src/operations/users/user-edit.tsx

```diff
@@ -0,0 +1,11 @@
+import { FC } from "react";
+import { Edit, SimpleForm, TextInput } from "react-admin";
+
+export const UserEdit: FC = () => {
+  <Edit>
+    <SimpleForm>
+      <TextInput source="name" label="Username" />
+      <TextInput source="email" label="Email" type="email"/>
```

Comment:
add validate={[required(), email()] on email input 
add validate={[required()]} on name input
You can remove type="email" 

import them like this: 

import {email} from "react-admin"
import { required } from "@/common/input-validator";